sos/plugins/openstack_glance.py

```diff
@@ -44,6 +46,18 @@ def setup(self):
 
         self.add_copy_spec("/etc/glance/")
 
+        run_cmds = False
+        for os_var in ['OS_USERNAME', 'OS_PASSWORD', 'OS_TENANT_NAME']:
+            if os_var not in os.environ:
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
+                run_cmds = True
```

Comment:
You can also avoid the explicit loop entirely by using a list comprehension and `all()`:
```python
vars = [p in os.environ for p in ['OS_USERNAME', 'OS_PASSWORD', 'OS_TENANT_NAME']]
if not all(vars):
    # freak out
```